src/OnlineSales/Entities/Domain.cs

```diff
@@ -36,6 +37,15 @@ public class Domain : BaseEntityWithIdAndDates
     public List<DnsRecord>? DnsRecords { get; set; }
 
     public bool? DnsCheck { get; set; }
+
+    public int? AccountId { get; set; }
+
+    [DeleteBehavior(DeleteBehavior.Restrict)]
+    [JsonIgnore]
+    [ForeignKey("AccountId")]
+    public virtual Account? Account { get; set; }
+
+    public bool AccountSynced { get; set; } = false;
```

Comment:
@WASandaruwan why do you need AccountSynced property for the domain? There is AccountId already and it could have null or id. Null means there is no linked account yet, Id means there is.

As described in the task the account should always be created, even it is just an account with simply a Name =  domain name. Then the information about the account should be store in the account table and I see there is the AccountSynced attribute in the Account DTOs as well. AccountSynced for the Domain makes no sense for me. Please explain.